src/ca/sigmsg.rs

```diff
@@ -549,15 +569,25 @@ impl SignedMessageTbsCrl {
         issuer_key: &PublicKey,
         when: Time,
     ) -> Result<(), ValidationError> {
-        if self.this_update > when || self.next_update < when {
-            Err(ValidationError)
-        } else {
+        if self.this_update > when {
+            Err(VerificationError::new(
+                "CRL thisUpdate time in the future"
+            ).into())
+        }
+        else if self.next_update < when {
+            Err(VerificationError::new(
+                "CRL nextUpdate time in the past"
+            ).into())
+        }
+        else {
             match self.authority_key_id {
                 None => Ok(()),
                 Some(aki) => if issuer_key.key_identifier() == aki {
                     Ok(())
                 } else {
-                    Err(ValidationError)
+                    Err(VerificationError::new(
+                        "Authority Key Identifer differs from issuer key"
```

Comment:
Should this text mention CRL? The ones above mention for example CRL.

Reply:
pushed proposal

Reply:
I expanded the acronym AKI again for, er, personal taste.